admin/class-purger.php

```diff
@@ -682,42 +1064,77 @@ public function purge_them_all() {
 		$this->_purge_all_posts();
 		$this->_purge_all_taxonomies();
 		$this->_purge_all_date_archives();
-		$this->log( __( 'Everthing purged!', 'nginx-helper' ) );
+		$this->log( __( 'Everything purged!', 'nginx-helper' ) );
+
 		return true;
+
 	}
 
+	/**
+	 * Purge cache on term edited.
+	 *
+	 * @param int    $term_id Term id.
+	 * @param int    $tt_id Taxonomy id.
+	 * @param string $taxon Taxonomy.
+	 *
+	 * @return bool
+	 */
 	public function purge_on_term_taxonomy_edited( $term_id, $tt_id, $taxon ) {
 
 		$this->log( __( 'Term taxonomy edited or deleted', 'nginx-helper' ) );
 
-		if ( 'edit_term' == current_filter() && $term = get_term( $term_id, $taxon ) ) {
+		$term = get_term( $term_id, $taxon );
+
+		if ( 'edit_term' === current_filter() && ! is_wp_error( $term ) && ! empty( $term ) ) {
```

Comment:
Take value of `current_filter` function in one variable and use it so that it won't be executed twice.